src/main/java/com/faforever/client/teammatchmaking/PartyMemberItemController.java

```diff
@@ -0,0 +1,167 @@
+    chatUser = new ChatChannelUser(player.getUsername(), chatService.getChatUserColor(player.getUsername()), false, player);
+
+    countryImageView.imageProperty().bind(createObjectBinding(() -> countryFlagService.loadCountryFlag(
+        StringUtils.hasText(player.getCountry()) ? "" : player.getCountry()).orElse(null), player.countryProperty()));
+
+    avatarImageView.visibleProperty().bind(player.avatarUrlProperty().isNotNull().and(player.avatarUrlProperty().isNotEmpty()));
+    avatarImageView.imageProperty().bind(createObjectBinding(() -> Strings.isNullOrEmpty(player.getAvatarUrl()) ? null : avatarService.loadAvatar(player.getAvatarUrl()), player.avatarUrlProperty()));
+    leagueImageView.setManaged(false);
+
+    clanLabel.visibleProperty().bind(player.clanProperty().isNotEmpty().and(player.clanProperty().isNotNull()));
+    clanLabel.textProperty().bind(createStringBinding(() -> Strings.isNullOrEmpty(player.getClan()) ? "" : String.format("[%s]", player.getClan()), player.clanProperty()));
+
+    usernameLabel.textProperty().bind(player.usernameProperty());
+
+    leagueLabel.textProperty().bind(createStringBinding(
+        () -> i18n.get("leaderboard.divisionName", RatingUtil.getLeaderboardRating(player)).toUpperCase(),
+        player.globalRatingMeanProperty())); // TODO: replace this with divisionproperty once it is available
+    gameCountLabel.textProperty().bind(createStringBinding(
+        () -> i18n.get("teammatchmaking.gameCount", player.getNumberOfGames()).toUpperCase(),
+        player.numberOfGamesProperty()));
+
+    BooleanBinding isDifferentPlayerBinding = playerService.currentPlayerProperty().isNotEqualTo(player);
+    kickPlayerButton.visibleProperty().bind(teamMatchmakingService.getParty().ownerProperty().isEqualTo(playerService.currentPlayerProperty()).and(isDifferentPlayerBinding));
+    kickPlayerButton.managedProperty().bind(kickPlayerButton.visibleProperty());
+    playerStatusImageView.managedProperty().bind(playerStatusImageView.visibleProperty());
+    crownLabel.visibleProperty().bind(teamMatchmakingService.getParty().ownerProperty().isEqualTo(player));
+    playerCard.pseudoClassStateChanged(LEADER_PSEUDO_CLASS, teamMatchmakingService.getParty().getOwner().equals(player));
+
+    playerStatusImageView.setImage(uiService.getThemeImage(UiService.CHAT_LIST_STATUS_PLAYING));
+    playerService.getPlayerForUsername(player.getUsername()).get().statusProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue == PlayerStatus.IDLE) {
+        Platform.runLater(() -> {
+          playerStatusImageView.setVisible(false);
+          playerCard.pseudoClassStateChanged(PLAYING_PSEUDO_CLASS, false);
+        });
+      } else {
+        Platform.runLater(() -> {
+          playerStatusImageView.setVisible(true);
+          playerCard.pseudoClassStateChanged(PLAYING_PSEUDO_CLASS, true);
+        });
+      }
```

Comment:
We need to make sure this listener is removed if a player leaves the party

Reply:
Everytime a partymember leaves or joines the player cards get rerendered, that means it clears all children from the pane and instantiates new controllers. I am not sure, if this also cleans up the controllers, or if we need to do some form of garbage collection